### DISCUSSION TITLE
Appindicator support

### DISCUSSION THREAD

**Discussion by Aleksanaa** [andyholmes/valent#403] [Feature Request]
If anyone's interested, keep track of this issue: https://github.com/AyatanaIndicators/libayatana-appindicator/issues/22

**Comment by andyholmes** [comment]
This is also a good time to let me know if there's any API needed to support this :slightly_smiling_face:

**Reply by andyholmes** [reply]
FYI, I've opened this as #431. General discussion and opinions can happen here, while any specific implementation stuff can go on the issue :slightly_smiling_face:

**Comment by kevinnlsamuel** [comment]
@andyholmes you had mentioned you don't like AppIndicator… how do you _intend_ Valent to be used? a full gtk window minimised at all times? or is there a flag to run it in the background as a daemon?

i personally find it distracting to have a window open when i'm only looking at the notifications from the app in question

**Reply by andyholmes** [reply]
I just close the window, Valent will always run in the background unless something like the desktop prevents it from running in the background.

**Reply by kevinnlsamuel** [reply]
oh! i had no idea. just tried it out. thank you ^_^

**Reply by kurtbahartr** [reply]
> I just close the window, Valent will always run in the background unless something like the desktop prevents it from running in the background.

Well... It doesn't work on Hyprland after I did a Super+Q, which is the equivalent of Alt+F4 on regular DEs and on Windows. I'll try enabling close button in GTK options, and see if what you meant with "close the window" is clicking on the close button, later today.

**Reply by andyholmes** [reply]
Unless you select the "Quit" option or press Ctrl+Q, Valent won't exit by itself. Other than that I don't have an answer for you :shrug:

**Reply by kurtbahartr** [reply]
> Unless you select the "Quit" option or press Ctrl+Q, Valent won't exit by itself.

Alright so, I just checked it again and it works just the way you stated here now, at least according to `pgrep`. Now I'm starting to be even more confused the more I dig into troubleshooting...

> Other than that I don't have an answer for you 🤷

**Reply by andyholmes** [reply]
What led you to believe the process had stopped?

Maybe this is just a natural assumption when someone closes the main window of an application and there's no indicator, but the devices will remain connected so it should be easy to infer?

**Reply by kurtbahartr** [reply]
> What led you to believe the process had stopped?
> 
> Maybe this is just a natural assumption when someone closes the main window of an application and there's no indicator, but the devices will remain connected so it should be easy to infer?

Not really. I did assume the process was running but then saw my phone disconnect from my laptop abruptly until I reopened the main window. I didn't think of checking `pgrep` at the time, so there's that too.

The actual issue was because of my university WiFi, which blocks any and all kinds of P2P connections (in efforts to block torrenting), IPv4 broadcasts and IPv6 Neighbor Discovery Protocol (in efforts to block potential threat actors that make use of either of these) through use of Fortinet. I tried again using my phone's mobile hotspot this time and it did stay connected - That was why I'm more confused.
